app/controllers/protocols_controller.rb

```diff
@@ -532,12 +532,60 @@ def import
   end
 
   def export
+    # Make a zip output stream and send it to the client
     respond_to do |format|
-      format.json {
-        render json: {
-            protocols: export_protocols(@protocols)
-          }, status: :ok
-      }
+      format.html do
+        z_output_stream = Zip::OutputStream.write_buffer do |ostream|
+          ostream.put_next_entry('scinote.xml')
+          ostream.print(generate_envelope_xml(@protocols))
+          ostream.put_next_entry('scinote.xsd')
+          ostream.print(generate_envelope_xsd)
+          ostream.put_next_entry('eln.xsd')
+          ostream.print(generate_eln_xsd)
+
+          # Create folder and xml file for each protocol and populate it
+          @protocols.each do |protocol|
+            protocol_dir = get_guid(protocol.id).to_s
+            ostream.put_next_entry("#{protocol_dir}/eln.xml")
+            ostream.print(generate_protocol_xml(protocol))
+            # Add assets to protocol folder
+            if protocol.steps.count > 0
```

Comment:
Use next so that the Hound will shut up :P. Basically, if an if statement is at the end, you should use next instead.